app/Http/Controllers/Admin/AnnouncementController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Models\Group;
+use Illuminate\Http\Request;
+use App\Models\Announcement;
+use App\Http\Controllers\Controller;
+use App\Repositories\Group\GroupRepository;
+use App\Http\Resources\Announcement as AnnouncementResource;
+use App\Http\Requests\AnnouncementRequest;
+
+class AnnouncementController extends Controller
+{
+    protected $groupRepository;
+
+    public function __construct(GroupRepository $groupRepository) {
+        $this->groupRepository = $groupRepository;
+    }
+    public function index($group_id)
+    {
+        $group = $this->groupRepository->findOrFail($group_id);
```

Comment:
Chỗ này with('announcements') mình tiết kiệm được câu truy vấn không chị nhỉ

Reply:
Dạ em mới học php nên chưa biết nhiều, mong anh giải thích thêm giúp em với ạ.

Reply:
Dạ ở đây em thấy $group = $this->groupRepository->findOrFail($group_id); vẫn phải chạy để trả $group về cho view nên khi chuyển đổi với with('announcements') vẫn không làm code gọn hơn được ạ

Reply:
mà chức năng này em không viết tại hàm show mà lại để ở hàm index

Reply:
Em nghĩ viết tại hàm show là không ổn, vì hàm show ở group controller có thể để show thông tin của 1 group, còn hàm show của announcement thì hiển thị thông tin của 1 announcemnet. Em đã chuyển announcement controller vào namespace group, hiển thị announcement theo từng group trong hàm index ở đây có vẻ ổn hơn ạ